pkg/server/cstorvolumeconfig/backup_endpoint.go

```diff
@@ -0,0 +1,545 @@
+	err = bOps.clientset.OpenebsV1alpha1().CStorBackups(ns).Delete(cstorBackup, &metav1.DeleteOptions{})
+	if err != nil && !k8serror.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to delete cstorbackup: %s resource", cstorBackup)
+	}
+	return nil
+}
+
+// backupCreateRequestValidations validates the backup create request
+func backupCreateRequestValidations(backUp *openebsapis.CStorBackup) error {
+	// backup name is expected
+	if len(strings.TrimSpace(backUp.Spec.BackupName)) == 0 {
+		return CodedError(400, fmt.Sprintf("Failed to create backup: missing backup name "))
+	}
+
+	// namespace is expected
+	if len(strings.TrimSpace(backUp.Namespace)) == 0 {
+		return CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing namespace", backUp.Spec.BackupName))
+	}
+
+	// volume name is expected
+	if len(strings.TrimSpace(backUp.Spec.VolumeName)) == 0 {
+		return CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing volume name", backUp.Spec.BackupName))
+	}
+
+	// backupIP is expected for remote snapshot
+	if !backUp.Spec.LocalSnap && len(strings.TrimSpace(backUp.Spec.BackupDest)) == 0 {
+		return CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing backupIP", backUp.Spec.BackupName))
+	}
+
+	// snapshot name is expected
+	if len(strings.TrimSpace(backUp.Spec.SnapName)) == 0 {
+		return CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing snapName", backUp.Spec.BackupName))
+	}
+	return nil
+}
+
+// getVolumeIP fetches the cstor target service IP Address
+func getVolumeIP(volumeName string, clientset clientset.Interface) (string, error) {
+	namespace := getOpenEBSNamespace()
+
+	// Fetch the corresponding cstorvolume
+	cstorvolume, err := clientset.CstorV1().CStorVolumes(namespace).
+		Get(volumeName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	return cstorvolume.Spec.TargetIP, nil
+}
+
+// getOpenEBSNamespace returns namespace where
+// cvc operator is running
+func getOpenEBSNamespace() string {
+	if openebsNamespace == "" {
+		openebsNamespace = util.GetEnv(util.OpenEBSNamespace)
+	}
+	return openebsNamespace
+}
+
+// findHealthyCVR returns Heathy CVR if exists else
+// it will return error
+func findHealthyCVR(
+	openebsClient clientset.Interface,
+	volume string) (*cstorapis.CStorVolumeReplica, error) {
+	namespace := getOpenEBSNamespace()
+	listOptions := metav1.ListOptions{
+		LabelSelector: "openebs.io/persistent-volume=" + volume,
+	}
+
+	cvrList, err := openebsClient.CstorV1().CStorVolumeReplicas(namespace).List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// Select a healthy cvr for backup
+	for _, cvr := range cvrList.Items {
+		if cvr.Status.Phase == cstorapis.CVRStatusOnline {
+			return &cvr, nil
+		}
+	}
+
+	return nil, errors.New("unable to find healthy CVR")
+}
+
+// getLastBackupSnap will fetch the last successful backup's snapshot name
+func (bOps *backupAPIOps) getLastBackupSnap(backUp *openebsapis.CStorBackup) (string, error) {
```

Comment:
This function is misleading in terms of what it does vs the comments and the function name.
ToDo: Next PR can improve this.